### PR TITLE
ElasticQuota: add unit test for PostFilter & PodEligibleToPreemptOthers

### PR DESCRIPTION
**Pull request by Gekko0114**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Added unit tests for PostFilter & PodEligibleToPreemptOthers in ElasticQuota.
I referred to the code of [default_preemption_test.go](https://github.com/kubernetes/kubernetes/blob/1af56548af167d836eff33a5f57552fa417cbc0b/pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go)

#### Special notes for your reviewer:
Thank you for reviewing this PR.
Please let me know if I have made any mistakes or if there are any improvements that could be made.
Your comments and feedback are appreciated.

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

**Comment by Gekko0114** [comment]
Hi @denkensk, @zwpaper, @Huang-Wei,

Could you please take a look when you have time? It's not urgent.
Thanks,

**Comment by Gekko0114** [comment]
Hi @denkensk , @ffromani , @Huang-Wei ,

Sorry for ping you multiple times.
Could you review and approve it?
I only added some unit tests and it has been almost a month since I created it, so I would like someone to review it.

**Comment by ffromani** [comment]
> Hi @denkensk , @ffromani , @Huang-Wei ,
> 
> Sorry for ping you multiple times. Could you review and approve it? I only added some unit tests and it has been almost a month since I created it, so I would like someone to review it.

Sorry, busy times. I need to learn a bit myself about ElasticQuota before I can add a meaningful review, this is the reason why is taking long on my side. I'll try to prioritize during this week.

File: pkg/capacityscheduling/capacity_scheduling_test.go [hunk 1]
```diff
@@ -168,6 +179,203 @@ func TestPreFilter(t *testing.T) {
 	}
 }
 
+func TestPostFilter(t *testing.T) {
+	res := map[v1.ResourceName]string{v1.ResourceMemory: "150"}
+	tests := []struct {
+		name                  string
+		pod                   *v1.Pod
+		pods                  []*v1.Pod
```
Comment (denkensk) [inline pkg/capacityscheduling/capacity_scheduling_test.go:187]:
```suggestion
		existPods                  []*v1.Pod
```

File: pkg/capacityscheduling/capacity_scheduling_test.go [hunk 2]
```diff
@@ -168,6 +179,203 @@ func TestPreFilter(t *testing.T) {
...
+					Used: &framework.Resource{
+						Memory: 100,
+					},
+				},
+			},
+			wantResult: framework.NewPostFilterResultWithNominatedNode("node-a"),
+			wantStatus: framework.NewStatus(framework.Success),
+		},
+		{
+			name: "Without elasticQuotas",
```
Comment (denkensk) [inline pkg/capacityscheduling/capacity_scheduling_test.go:281]:
```suggestion
			name: "without elasticQuotas",
```

**Comment by Gekko0114** [comment]
Sure, thank you so much. I appreciate it :)

File: pkg/capacityscheduling/capacity_scheduling_test.go [hunk 3]
```diff
@@ -602,6 +810,222 @@ func TestDryRunPreemption(t *testing.T) {
...
+					},
+					Used: &framework.Resource{
+						Memory: 100,
+					},
+				},
+			},
+			expected: false,
+		},
+		{
+			name: "Pod without ElasticQuotas  and terminating pod exists",
```
Comment (denkensk) [inline pkg/capacityscheduling/capacity_scheduling_test.go:937]:
```suggestion
			name: "Pod without elasticQuotas and terminating pod exists in the same namespace",
```

File: pkg/capacityscheduling/capacity_scheduling_test.go [hunk 4]
```diff
@@ -602,6 +810,222 @@ func TestDryRunPreemption(t *testing.T) {
...
+					},
+				},
+			},
+			expected: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			registeredPlugins := []st.RegisterPluginFunc{
```
Comment (denkensk) [inline pkg/capacityscheduling/capacity_scheduling_test.go:979]: Can you define a function to do this? And we can reuse it in other places.

File: pkg/capacityscheduling/capacity_scheduling_test.go [hunk 5]
```diff
@@ -602,6 +810,222 @@ func TestDryRunPreemption(t *testing.T) {
...
+					},
+					Used: &framework.Resource{
+						Memory: 0,
+					},
+				},
+			},
+			expected: false,
+		},
+		{
+			name: "Pod with nominated node",
```
Comment (denkensk) [inline pkg/capacityscheduling/capacity_scheduling_test.go:849]:
```suggestion
			name: "Pod with unschedulableAndUnresolvable nominated node",
```

File: pkg/capacityscheduling/capacity_scheduling_test.go [hunk 6]
```diff
@@ -602,6 +810,222 @@ func TestDryRunPreemption(t *testing.T) {
 	}
 }
 
+func TestPodEligibleToPreemptOthers(t *testing.T) {
+	res := map[v1.ResourceName]string{v1.ResourceMemory: "150"}
+	tests := []struct {
+		name                string
+		pod                 *v1.Pod
+		pods                []*v1.Pod
```
Comment (denkensk) [inline pkg/capacityscheduling/capacity_scheduling_test.go:818]: also existPods

File: pkg/capacityscheduling/capacity_scheduling_test.go [hunk 7]
```diff
@@ -602,6 +810,222 @@ func TestDryRunPreemption(t *testing.T) {
...
+					},
+					Used: &framework.Resource{
+						Memory: 0,
+					},
+				},
+			},
+			expected: false,
+		},
+		{
+			name: "Pod with terminating pod in the different namespace",
```
Comment (denkensk) [inline pkg/capacityscheduling/capacity_scheduling_test.go:899]:
```suggestion
			name: "Pod with terminating pod in another namespace",
```

File: pkg/capacityscheduling/capacity_scheduling_test.go [hunk 8]
```diff
@@ -602,6 +810,222 @@ func TestDryRunPreemption(t *testing.T) {
...
+			},
+			nodes: []*v1.Node{
+				st.MakeNode().Name("node-a").Capacity(res).Obj(),
+			},
+			nominatedNodeStatus: nil,
+			elasticQuotas:       map[string]*ElasticQuotaInfo{},
+			expected:            false,
+		},
+		{
+			name: "Pod with other namespace's ElasticQuotas and terminating pod exists",
```
Comment (denkensk) [inline pkg/capacityscheduling/capacity_scheduling_test.go:950]:
```suggestion
			name: "Pod without ElasticQuotas and terminating pod exists in another namespace",
```

File: pkg/capacityscheduling/capacity_scheduling_test.go [hunk 9]
```diff
@@ -460,7 +668,7 @@ func TestDryRunPreemption(t *testing.T) {
 		},
 		{
 			name: "cross-namespace preemption",
-			pod:  makePod("t1-p", "ns1", 50, 0, 0, highPriority, "", "t1-p"),
+			pod:  makePod("t1-p", "ns1", 50, 0, 0, highPriority, "t1-p", ""),
```
Comment (denkensk) [inline pkg/capacityscheduling/capacity_scheduling_test.go:671]: Thanks. Good catch

**Review by denkensk — COMMENTED** [review]
lgtm overall 
Just some small comments. Thanks @Gekko0114

**Comment by denkensk** [comment]
> Hi @denkensk , @ffromani , @Huang-Wei ,
> 
> Sorry for ping you multiple times. Could you review and approve it? I only added some unit tests and it has been almost a month since I created it, so I would like someone to review it.

Sorry for the late reply.

**Comment by Gekko0114** [comment]
Hi @denkensk ,

Thanks for your comments! I fixed it.
Please take a look when you have time.

**Comment by denkensk** [comment]
/lgtm
/approve 

Thanks for your contribution. @Gekko0114

**Pull request merged**